file.go

```diff
@@ -78,32 +78,41 @@ func extractFiles(zipFilePath, filesToExtractFromZipPath, localPath string) erro
 	// Add the path from which we will extract files to the path prefix so we can exclude the appropriate files
 	pathPrefix = filepath.Join(pathPrefix, filesToExtractFromZipPath)
 
-	// Iterate through the files in the archive,
-	// printing some of their contents.
+	// Iterate through the files in the archive.
 	for _, f := range r.File {
 
 		// If the given file is in the filesToExtractFromZipPath, proceed
 		if strings.Index(f.Name, pathPrefix) == 0 {
 
+			path := filepath.Join(localPath, strings.TrimPrefix(f.Name, pathPrefix))
+
 			if f.FileInfo().IsDir() {
-				// Create a directory
-				os.MkdirAll(filepath.Join(localPath, strings.TrimPrefix(f.Name, pathPrefix)), 0777)
+				os.MkdirAll(path, 0777)
 			} else {
-				// Read the file into a byte array
-				readCloser, err := f.Open()
+				err = writeFileFromZip(f, path)
 				if err != nil {
-					return fmt.Errorf("Failed to open file %s: %s", f.Name, err)
+					return err
 				}
+			}
+		}
+	}
 
-				byteArray, err := ioutil.ReadAll(readCloser)
-				if err != nil {
-					return fmt.Errorf("Failed to read file %s: %s", f.Name, err)
-				}
+	// Sym links may refer to files within the repo, in which case we first need to copy all files, and then process symlinks
+	if fetchOptions != nil && fetchOptions.ResolveSymlinks {
+		fmt.Println("--resolve-symlinks is set to true, so replacing symlinks with the contents of their targets")
 
-				// Write the file
-				err = ioutil.WriteFile(filepath.Join(localPath, strings.TrimPrefix(f.Name, pathPrefix)), byteArray, 0644)
-				if err != nil {
-					return fmt.Errorf("Failed to write file: %s", err)
+		for _, f := range r.File {
```

Comment:
`file`. When you're 3 lines lower, `f` is meaningless.

Reply:
This was there from before, but it's a fair point. Updated.